pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2020 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+)
+
+// Important:
+// The procedure working with this file:
+// 1. Edit the file
+// 1. Run "make generate" to regenerate code
+// 2. Run "make manifests" to regenerate the CRD
+
+func init() {
+	SchemeBuilder.Register(&AtlasProject{}, &AtlasProjectList{})
+}
+
+type ScopeType string
+
+const (
+	ClusterScopeType  ScopeType = "CLUSTER"
+	DataLakeScopeType ScopeType = "DATA_LAKE"
+)
+
+// AtlasDatabaseUserSpec defines the desired state of Database User in Atlas
+type AtlasDatabaseUserSpec struct {
+	// Project is a reference to AtlasProject resource the user belongs to
+	Project ResourceRef `json:"projectRef"`
+
+	// DatabaseName is a Database against which Atlas authenticates the user. Must be equal to 'admin'
+	// as only SCRAM authentication is supported so far.
+	DatabaseName string `json:"databaseName"`
+
+	// DeleteAfterDate is a timestamp in ISO 8601 date and time format in UTC after which Atlas deletes the user.
+	// The specified date must be in the future and within one week of the time you make the API request.
```

Comment:
I think this comment has a little too much implementation detail leaked in, I think the only information necessary for the user is that `The specified date must be in the future and within one week`, the API request is the operator's job to handle!